src/Simplex/Chat.hs

```diff
@@ -462,25 +462,37 @@ processChatCommand = \case
     editedItemId <- withStore $ \st -> getGroupChatItemIdByText st user groupId (Just localDisplayName) (safeDecodeUtf8 editedMsg)
     let mc = MCText $ safeDecodeUtf8 msg
     processChatCommand $ APIUpdateChatItem CTGroup groupId editedItemId mc
+  -- old file protocol
   SendFile cName f -> withUser $ \user@User {userId} -> withChatLock $ do
     (fileSize, chSize) <- checkSndFile f
     contact <- withStore $ \st -> getContactByName st userId cName
-    (agentConnId, connReq) <- withAgent (`createConnection` SCMInvitation)
-    let fileInv = FileInvitation {fileName = takeFileName f, fileSize, fileConnReq = ACR SCMInvitation connReq}
+    (agentConnId, fileConnReq) <- withAgent (`createConnection` SCMInvitation)
+    let fileInv = FileInvitation {fileName = takeFileName f, fileSize, fileConnReq = Just fileConnReq}
     SndFileTransfer {fileId} <- withStore $ \st ->
       createSndFileTransfer st userId contact f fileInv agentConnId chSize
     ci <- sendDirectChatItem user contact (XFile fileInv) (CISndFileInvitation fileId f) Nothing
     withStore $ \st -> updateFileTransferChatItemId st fileId $ chatItemId' ci
     setActive $ ActiveC cName
     pure . CRNewChatItem $ AChatItem SCTDirect SMDSnd (DirectChat contact) ci
+  -- new file protocol
+  SendFileInv cName f -> withUser $ \user@User {userId} -> withChatLock $ do
+    (fileSize, chSize) <- checkSndFile f
+    contact <- withStore $ \st -> getContactByName st userId cName
+    let fileInv = FileInvitation {fileName = takeFileName f, fileSize, fileConnReq = Nothing}
+    fileId <- withStore $ \st -> createSndFileTransferV2 st userId contact f fileInv chSize
+    ci <- sendDirectChatItem user contact (XFile fileInv) (CISndFileInvitation fileId f) Nothing
```

Comment:
this will have to change for images somehow, as we won't have a separate chat item for file, correct?